django/core/management/base.py

```diff
@@ -365,7 +374,11 @@ def execute(self, *args, **options):
             self.stderr = OutputWrapper(options['stderr'])
 
         if self.requires_system_checks and not options['skip_checks']:
-            self.check()
+            tags = None
+            if isinstance(self.requires_system_checks, list):
+                tags = self.requires_system_checks
+            self.check(tags=tags)
```

Comment:
We should validate `self.requires_system_checks`, currently we run all checks also for any truthy value e.g. `requires_system_checks = 'x'`, maybe
```python
if not isinstance(value, (list, tuple)) and value != '__all__':
    raise TypeError(...)
```

Reply:
Validation added and a test added to check raising of the `TypeError`